### PR TITLE
Make TranscodingJobHelper an IDisposable (api-migration)

### PR DESCRIPTION
**Pull request by Ullmie02**
**Changes**
~~This adds an `IServerEntryPoint` to `Jellyfin.Api`. This class is needed because the remaining services require a list of all transcoding jobs which are currently running.~~
- Adds event listeners to `sessionManager.PlaybackProgress` and `sessionManager.PlaybackStart`.
- Makes `TranscodingJobHelper` an `IDisposable` so the event listeners get removed.

### CODE REVIEW

**Comment by crobibero** [comment]
Looks like we had duplicate efforts! Some of the TranscodingJob methods were converted as part of https://github.com/jellyfin/jellyfin/pull/3460

**Comment by Ullmie02** [comment]
Yeah, and I even reviewed it. My bad. But is the list of the active transcoding jobs at the right place or should it be implemented differently? Depending on that, I would either remove the duplicate methods (and probably move some to TranscodingHelpers.cs) or completely close this PR

**Comment by crobibero** [comment]
I think the TranscodingHelper now needs to implement `IDisposable` so `sessionManager.PlaybackProgress` gets reset,
but I'm not sure how this will all work as the class scope has changed with the move to webapi. We might need to somehow make a singleton TranscodingHelper that the controllers can use

File: Jellyfin.Api/Helpers/TranscodingJobHelper.cs [hunk 1]
```diff
@@ -350,5 +361,30 @@ private void DeleteHlsPartialStreamFiles(string outputFilePath)
...
+            if (!string.IsNullOrWhiteSpace(e.PlaySessionId))
+            {
+                PingTranscodingJob(e.PlaySessionId, e.IsPaused);
+            }
+        }
+
+        /// <summary>
+        /// Deletes the encoded media cache.
+        /// </summary>
+        private void DeleteEncodedMediaCache()
```
Comment (cvium) [inline Jellyfin.Api/Helpers/TranscodingJobHelper.cs:376]: This is unused?

Comment (crobibero) [reply Jellyfin.Api/Helpers/TranscodingJobHelper.cs:376]: I believe this PR is to bring the remaining functions over to Jellyfin.Api that many other controllers rely on

Comment (Ullmie02) [reply Jellyfin.Api/Helpers/TranscodingJobHelper.cs:376]: This function was called on server startup but I haven't found a way to call it yet. 
The main goal of this PR was to bring the remaining functions to Jellyfin.Api. But I think I created more confusion than I helped.

**Pull request merged**